assets/js/googlesitekit/modules/datastore/settings.js

```diff
@@ -150,6 +150,22 @@ export const selectors = {
 		const storeName = select( CORE_MODULES ).getModuleStoreName( slug );
 		return !! select( storeName )?.canSubmitChanges?.();
 	} ),
+
+	/**
+	 * Checks whether there are changes to save for a module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} slug Slug for module store.
+	 * @return {boolean} Whether changes have been made in the settings form.
+	 */
+	haveSettingsChanged: createRegistrySelector(
+		( select ) => ( state, slug ) => {
+			invariant( slug, 'slug is required.' );
+			const storeName = select( CORE_MODULES ).getModuleStoreName( slug );
+			return !! select( storeName )?.haveSettingsChanged?.();
+		}
+	),
```

Comment:
We have a generic [`haveSettingsChanged`](https://github.com/google/site-kit-wp/blob/cde25db0ac413e2345df5087c2662b2c86c0e454/assets/js/googlesitekit/data/create-settings-store.js#L288) selector already that takes different arguments, why use the exact same name?

That's a bit confusing for me, maybe this one could be `haveSettingsChangedForModule`, etc.?

Though I'm not sure why we wouldn't use `select( MODULES_ADS ).haveSettingsChanged()` directly, above https://github.com/google/site-kit-wp/pull/8545/files#diff-35a209af9b8af36a7078eba599222127fa6ef4c9bdaa8b38b5dfecfdd685a2ceR66

Reply:
@tofumatt In terms of the selector naming I was following the naming of the `canSubmitChanges` selector above which shares the naming of the underlaying module selector:

https://github.com/google/site-kit-wp/pull/8545/files#diff-80a2c5f54e6197f4fad0b4cf508c8337f5af754bf8ddd8b6286f65db5b020442R148-R152

The reason we can't use the module level selector is that the `assets/js/components/settings/SettingsActiveModule/Footer.js` component is shared between each module settings, not just the Ads module.